File: others/generate_excel.pl
use Win32::OLE;
# Start Excel and make it visible
$xlApp = Win32::OLE->new('Excel.Application');
$xlApp->{Visible} = 0;

# Create a new workbook
$xlBook = $xlApp->Workbooks->Add;

# Our data that we will add to the workbook...
$mydata = [["Item",     "Category", "Price"], 
           ["Nails",    "Hardware",  "5.25"],
           ["Shirt",    "Clothing", "23.00"],
           ["Hammer",   "Hardware", "16.25"],
           ["Sandwich", "Food",      "5.00"],
           ["Pants",    "Clothing", "31.00"],
           ["Drinks",   "Food",      "2.25"]];

# Write all the data at once...
$rng = $xlBook->workSheets(1)->Range("A1:C7");
$rng->{Value} = $mydata;

# Wait for user input...
print "Press <return> to continue...";
$x = <STDIN>;

# Clean up

$xlBook->saveas('C:\Users\jayhsieh\Desktop\abc.xlsx');
$xlApp->Quit;
$xlBook = 0;
$xlApp = 0;

print "All done.";

File: others/generate_excel.py
import win32com.client
import os

xlsApp = win32com.client.Dispatch("Excel.Application")
xlsApp.Visible = 0                  #顯示 Excel
xlsBook = xlsApp.Workbooks.Add()    #新增一工作簿
xlsSheet = xlsBook.Worksheets(1)  #新增的工作簿預設含三個工作表
xlsSheet.Cells(1,1).Value=100
xlsBook.SaveAs('C:\\Users\\jayhsieh\\Desktop\\text.xlsx')
xlsBook.Close(SaveChanges=1)
xlsApp.Quit()
del xlsApp
